pkg/framework/plugins/nodeutilization/lownodeutilization.go

```diff
@@ -102,22 +106,15 @@ func (l *LowNodeUtilization) Name() string {
 
 // Balance extension point implementation for the plugin
 func (l *LowNodeUtilization) Balance(ctx context.Context, nodes []*v1.Node) *frameworktypes.Status {
-	nodeUsage, err := getNodeUsage(nodes, l.resourceNames, l.handle.GetPodsAssignedToNodeFunc())
-	if err != nil {
+	if err := l.usageSnapshot.capture(nodes); err != nil {
```

Comment:
IMO it would be better not to store the data in the usageSnapshot/usageClient (the naming is a bit confusing). And return a new struct when we do the capture. Then we can pass the the struct around without unnecessarily binding it with the client.

Reply:
The current implementation of the client based on kubernetes metrics does not capture pod usage. It reads the usage as low node utilization plugin goes and asks for it. Which would make the captured snapshot only partial.

Reply:
I agree using a word snapshot does not fully correspond to the intention. Would `sync` work instead of `capture`?

Reply:
I will rename the variable to usageClient. I would like to avoid any splitting atm. I don't think it's worth doing it given the usage client is not expected to be exported elsewhere.

Both low/high node utilization plugins are expected to invoke capture/sync right after `Balance` method is invoked. There's no cache invalidation as each `Balance` is expected to get the latest data.

Reply:
+1, and it would be good to document that in the interface

Reply:
Done